src/libraries/System.Private.CoreLib/src/System/Reflection/TypeDelegator.cs

```diff
@@ -57,6 +57,8 @@ public TypeDelegator([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.
         public override string? AssemblyQualifiedName => typeImpl.AssemblyQualifiedName;
         public override Type? BaseType => typeImpl.BaseType;
 
+        public override int GetArrayRank() => typeImpl.GetArrayRank();
```

Comment:
This was added to allow Xunit `[MemberData()]` to be used for testability. 

I don't see a reason why to not do this.